packages/shorebird_cli/lib/src/archive_analysis/archive_differ.dart

```diff
@@ -1,19 +1,75 @@
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
 import 'package:shorebird_cli/src/archive_analysis/archive_analysis.dart';
 
 /// Thrown when an [ArchiveDiffer] fails to generate a [FileSetDiff].
 class DiffFailedException implements Exception {}
 
 /// Computes content differences between two archives.
 abstract class ArchiveDiffer {
+  /// Asset files that are not considered to be breaking changes.
+  static const assetFileNamesToIgnore = {
+    'AssetManifest.bin',
+    'AssetManifest.json',
+    'NOTICES.Z',
+  };
+
   /// Files that have been added, removed, or that have changed between the
   /// archives at the two provided paths.
   FileSetDiff changedFiles(String oldArchivePath, String newArchivePath);
 
   /// Whether there are asset differences between the archives that may cause
   /// issues when patching a release.
-  bool containsPotentiallyBreakingAssetDiffs(FileSetDiff fileSetDiff);
+  bool containsPotentiallyBreakingAssetDiffs(FileSetDiff fileSetDiff) {
+    final assetsDiff = assetsFileSetDiff(fileSetDiff);
+
+    // If assets were added, we need to warn the user about asset differences.
+    if (assetsDiff.addedPaths.isNotEmpty) {
+      return true;
+    }
+
+    return assetsDiff.changedPaths
```

Comment:
Given you've repeated this complex whereNot twice, I might have written this:

return removeIgnoredFiles(assetsDiff.changedPaths).isEmpty() or similar.